sentry-core/src/futures.rs

```diff
@@ -0,0 +1,96 @@
+use std::future::Future;
```

Comment:
Might be worth adding a module-level docstring with the common example of using `FutureExt::bind_hub`?

Reply:
I hope I can clean up all the docs etc in a followup.